overlord/devicestate/devicemgr.go

```diff
@@ -782,6 +782,20 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// canAutoRefresh is a helper that checks if the device is able to
+// auto-refresh
+func canAutoRefresh(st *state.State) bool {
+	// no need to wait for seeding on classic
+	if release.OnClassic {
+		return true
+	}
+
+	// on all-snap devices we need to be seeded first
+	var seeded bool
+	st.Get("seeded", &seeded)
+	return seeded
+}
```

Comment:
Indeed, it is a really good question. I lean towards letting it try even when device registration is not done yet, but can change it to wait for that (or implement the logic you mentioned, i.e. try refresh only after e.g. 24h after device reg was tried first).